### PR TITLE
AWS cluster definition

### PR DESCRIPTION
**Pull request by marians**
This PR aims to spec the API changes required by

- AWS instance type selection https://github.com/giantswarm/giantswarm/issues/1310
- AWS resource tagging https://github.com/giantswarm/giantswarm/issues/1318

### CODE REVIEW

**Comment by xh3b4sd** [comment]
I would like to properly separate provider specific configurations. The current state of the PR seems to suggest to mix up Giantswarm specific information with AWS specific information. I don't think this is the right thing. So it would be cool if we could have dedicated AWS blocks where AWS specific configuration has to be provided.

**Comment by marians** [comment]
@xh3b4sd So like `aws.tags` or even like `provider.aws.tags`?

**Comment by xh3b4sd** [comment]
Something along these lines would be better than having `really_long_property_names` all mixed together with other information, yes. IMO a clean logical separation helps working with these structures.

**Comment by marians** [comment]
@xh3b4sd If you like, please give it another look. Better?

File: details/CLUSTER_DEFINITION.md [hunk 1]
```diff
@@ -43,14 +43,92 @@ __Note:__ upon cluster creation, some of the attributes shown below MUST NOT be
...
+        {
+            "aws": {
+              "instance_type": "r3.xlarge"
+            },
+            "labels": {
+                "nodetype": "big-ram"
+            }
+        }
+    ]
+}
```
Comment (xh3b4sd) [inline details/CLUSTER_DEFINITION.md]: I like that a lot better. Thanks. The object below is still the older version though.

Comment (marians) [inline details/CLUSTER_DEFINITION.md]: Fixed. Thanks!

**Comment by marians** [comment]
@xh3b4sd Sicne you approved already, there are two significant changes:

1. https://github.com/giantswarm/api-spec/pull/10/commits/3b2c036e03453183de368aa256ca2b46bc010323 changes the documentation so that the added `aws.*` attributes are explained.
2. https://github.com/giantswarm/api-spec/pull/10/commits/59fd2ddd465de93e92ea7aeedeb07e1b9e92577a changes the spec YAML to add the attributes.

**Comment by marians** [comment]
TODO: 
- make storage.size_gb a number
- descripe instance type constraints per installation

**Comment by oponder** [comment]
This is waiting on tags to be implemented / further discussed

@marians will see what can be merged and what is still waiting implementation

**Comment by marians** [comment]
FYI: Merging this after I removed the part regarding the cluster-level aws settings (resource tags), wihch are moved to a new PR:

**Pull request merged**